// antd-mobile/lib/steps/index.web.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports["default"] = undefined;

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _rcSteps = require('rc-steps');

var _rcSteps2 = _interopRequireDefault(_rcSteps);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var __assign = undefined && undefined.__assign || Object.assign || function (t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) {
            if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
        }
    }
    return t;
};

function isString(str) {
    return typeof str === 'string';
}

var Steps = function (_React$Component) {
    (0, _inherits3["default"])(Steps, _React$Component);

    function Steps() {
        (0, _classCallCheck3["default"])(this, Steps);
        return (0, _possibleConstructorReturn3["default"])(this, _React$Component.apply(this, arguments));
    }

    Steps.prototype.render = function render() {
        var _this2 = this;

        var _props = this.props,
            current = _props.current,
            direction = _props.direction;

        return _react2["default"].createElement(_rcSteps2["default"], __assign({}, this.props, { direction: direction }), this.props.children.map(function (item, index) {
            var errorTail = -1;
            if (index < _this2.props.children.length - 1) {
                var status = _this2.props.children[index + 1].props.status;
                if (status === 'error') {
                    errorTail = index;
                }
            }
            var errorTailCls = errorTail > -1 ? 'error-tail' : '';
            var iconName = void 0;
            var className = void 0;
            if (!!item.props.icon) {
                iconName = item.props.icon;
                if (isString(iconName)) {
                    className = '';
                    if (index > 0 && index <= current) {
                        iconName = 'check-circle';
                    } else if (item.props.status === 'error') {
                        iconName = 'cross-circle';
                    } else if (item.props.status === 'process') {
                        iconName = 'check-circle';
                    }
                }
            } else {
                className = index <= current ? null : 'ellipsis-item';
                if (index <= current) {
                    iconName = 'check-circle-o';
                } else if (item.props.status === 'error') {
                    iconName = 'cross-circle-o';
                } else {
                    iconName = 'ellipsis';
                }
            }
            className = errorTailCls + ' ' + className;
            return _react2["default"].cloneElement(item, { key: index, icon: iconName, className: className });
        }));
    };

    return Steps;
}(_react2["default"].Component);

exports["default"] = Steps;

Steps.Step = _rcSteps2["default"].Step;
Steps.defaultProps = {
    prefixCls: 'am-steps',
    iconPrefix: 'ant',
    labelPlacement: 'vertical',
    current: 0,
    direction: 'vertical'
};
module.exports = exports['default'];
